Improve the getting started docs

davidfischer opened the pull request:
* Adds an introductory few paragraphs to `docs/index` page to explain the value proposition of Read the Docs. The goal here is to answer the "Why Read the Docs" question.
* Breaks the getting started guide into 3 docs
  - getting started with Sphinx
  - getting started with MkDocs (this is brand new)
  - importing your documentation
* In the getting started with X guides, I briefly explain the benefits/trade-offs of using Sphinx/MkDocs
* Puts "connecting your account" (to GitHub, Bitbucket, GitLab) into a separate doc rather than in the getting started guide. This won't be as necessary going forward as more people choose to sign up with GitHub, etc.
* In the toctree, moves:
  - the "about Read the Docs" section below "Feature documentation"
  - puts the "Design" section into the developer documentation

## After merging

Create a redirect in the Read the Docs project dashboard for the "docs" project from `getting-started.html` -> `intro/getting-started-with-sphinx.html`.

## Notes

I would like to put full sentence/paragraph descriptions on different sections (eg. User documentation, Feature documentation, About Read the Docs, etc.), but there isn't a great way to do this. I still think we should do it to explain the differences between those various sections. Currently the best path I have for that is similar to what I did with "First steps" which is to make the toctree hidden and then also have duplicate links to the documents.

What are peoples' thoughts on having more screenshots in our docs? I put one in the import guide but I could easily create a lot more. I think especially in getting started guides, these are very reassuring and show people they are doing the right thing in the right place.

Fixes #4648

stsewd commented on docs/intro/getting-started-with-mkdocs.rst:
```diff
@@ -0,0 +1,66 @@
+
+.. sourcecode:: bash
+
+    $ pip install mkdocs
```
I'm not sure how this is rendered, but maybe dropping the `$` makes easier to copy/paste

davidfischer replied:
It doesn't really render any different from the [existing getting started guide](https://docs.readthedocs.io/en/latest/getting_started.html#in-restructuredtext). The only difference is that any shell builtins (eg. `echo`, `cd`, etc.) are syntax highlighted. It also is more explicit.

There's a few places in our docs that use the pattern of `$ shellcommand` so I'm hesitant to change it without changing it everywhere.

ericholscher replied:
I generally dislike the `$` for the reason mentioned (hard to copy/paste), but don't feel strongly.

stsewd commented on docs/intro/getting-started-with-mkdocs.rst:
```diff
@@ -0,0 +1,66 @@
+Open up http://127.0.0.1:8000/ in your web browser to see your documentation.
+You can make changes to your Markdown files and your docs will automatically rebuild.
+
+Once you have your documentation in a public repository such GitHub, Bitbucket, or GitLab,
```
I'm not an English guy, but shouldn't be `such as`?

stsewd commented on docs/intro/getting-started-with-sphinx.rst:
```diff
@@ -0,0 +1,128 @@
+.. sourcecode:: bash
+
+    $ cd /path/to/project
+    $ mkdir docs
```
We don't mention this on the MkDocs guide

davidfischer replied:
It also creates the `docs` directory if it doesn't exist.

stsewd commented on docs/intro/getting-started-with-sphinx.rst:
```diff
@@ -0,0 +1,128 @@
+
+.. sourcecode:: bash
+
+    $ make html
```
Maybe mentioning that this is only valid for unix like systems?

davidfischer commented:
> Shouldn't we mention the configuration part here? Like using a configuration file or the web admin.

I think that's a great idea.

Where do you think it should go? Perhaps just a sentence or two in "Importing Your Documentation >> Building your documentation" and it links to the [Yaml config docs](https://docs.readthedocs.io/en/latest/yaml-config.html)?

I also think the Yaml config docs should be linked from the [builds docs](https://docs.readthedocs.io/en/latest/builds.html).

stsewd commented:
> I also think the Yaml config docs should be linked from the builds docs.

Definitely, that's a good place

> Where do you think it should go? Perhaps just a sentence or two in "Importing Your Documentation >> Building your documentation" and it links to the Yaml config docs?

That's a good question p: not sure, maybe it makes sense on an early step? Looks like an intermediate step between creating the documentation and importing/building it on rtd.

davidfischer commented:
If you don't have a strong opinion, I lean toward putting it in "Importing Your Documentation >> Building your documentation". I think that because many projects (especially simple ones) won't need it and I want to keep the getting started guide short and simple.

7mobile56 commented:
2018年9月29日 07:36於 "Santos Gallegos" <notifications@github.com>寫道：

> I also think the Yaml config docs should be linked from the builds docs.
>
> Definitely, that's a good place
>
> Where do you think it should go? Perhaps just a sentence or two in
> "Importing Your Documentation >> Building your documentation" and it links
> to the Yaml config docs?
>
> That's a good question p: not sure, maybe it makes sense on an early step?
> Looks like an intermediate step between creating the documentation and
> importing/building it on rtd.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/rtfd/readthedocs.org/pull/4676#issuecomment-425594297>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AZ510P5mRLHbWm0Q7cRn1Q6eumpv693gks5ufrKOgaJpZM4XALPy>
> .
>

davidfischer commented:
I linked to the Yaml config on "Importing Your Documentation". The builds page needs a bit more work and is a little out of scope for this PR. I created https://github.com/rtfd/readthedocs.org/issues/4679 to handle it.

ericholscher commented on docs/connected-accounts.rst:
```diff
@@ -0,0 +1,15 @@
+* Log into Read the Docs with your GitHub, Bitbucket, or GitLab credentials
+
+To connect your account, go to your *Settings* dashboard and select *Connected
+Services*. From here, you'll be able to connect to your GitHub, Bitbucket or GitLab
```
I believe you can also just "login" with the accounts and it will connect them. This is probably the most explicit way of documenting doing it though 👍

davidfischer replied:
Signing up will connect them. I'll make that a bit more clear.

davidfischer commented:
> I like screenshots. We don't have a great way of managing them, but I'm +1 on having them, especially in tutorial/intro docs.

I added screenshots. I moved the images around (under `_static`) and re-enabled `html_static_path` so that I could create hyperlinks to the images themselves.

davidfischer commented:
The lint errors are due to search stuff, not these docs.

ericholscher commented on docs/index.rst:
```diff
@@ -105,6 +124,8 @@ Information about development is also available:
    issue-labels
    security
+   design
+   RTD Theme <https://sphinx-rtd-theme.readthedocs.io/en/latest/>
```
We should make it a [subproject](https://docs.readthedocs.io/en/latest/subprojects.html) and use some of our fancier features that we build for users on our own stuff.

agjohnson commented on docs/intro/getting-started-with-mkdocs.rst:
```diff
@@ -0,0 +1,73 @@
+Open up http://127.0.0.1:8000/ in your web browser to see your documentation.
+You can make changes to your Markdown files and your docs will automatically rebuild.
+
+.. figure:: ../_static/images/first-steps/mkdocs-hello-world.png
```
`_static` is the theme static file output directory, and it isn't necessary to put `/images/` under `/_static`. It can exist top level

davidfischer replied:
It doesn't **have** to be but it is easier if we do put it in `_static`. Putting images in `_static` gives us a deterministic place to link to with `:target:` and the `:target:` will be the same as the source.

Without it, putting the images in `_images` or somewhere else means the `:target:` would need to be set to the directory (not `_images`) where the images get copied to by Sphinx during generation which is not intuitive at all.

agjohnson commented on docs/intro/getting-started-with-sphinx.rst:
```diff
@@ -0,0 +1,135 @@
+    and many projects large and small use Sphinx
+    including the Django project and the Python programming language.
+    However, Read the Docs also supports :doc:`MkDocs </intro/getting-started-with-mkdocs>`,
+    another tool for writing and building documentation.
```
I don't think we necessarily want to point out mkdocs here, if anything because we're not giving a reason for the user to consider mkdocs with this sentence. I think our position is that the author would be better off with Sphinx and RTD anyways, due to the lack of feature parity on mkdocs. I'd be :+1: on dropping the mkdocs callout here.

davidfischer replied:
I removed it. We could also discuss some of the differences like lack of PDF/e-reader support in mkdocs but I don't know if it is necessary.
